### DISCUSSION TITLE
Can't see command in debug mode (Custom Extension Development)

### DISCUSSION THREAD

**Discussion by pablojakub** [microsoft/vscode-discussions#892] [Extension Development QnA] [answered]
Hi,
I have just started my journey with custom extension development. I followed prompts on VSCode Webiste Docs - how to debug extension: 
https://code.visualstudio.com/api/get-started/your-first-extension#debugging-the-extension

Unfortunately when I'm running F5 on extension file, new window is being open but nothing is printed in console and I don't have my commands in prompt. When I set breakpoint on my extension file it turns out that it doesn't even hit it, as though it didn't open the file. 
I added also "VS Code Extension Development" in launch.json but also without any effect. My two configurations object looks like this: 

`{
			"args": [
				"--extensionDevelopmentPath=${workspaceFolder}"
			],
			"name": "Launch Extension",
			"outFiles": [
				"${workspaceFolder}/out/**/*.js"
			],
			"preLaunchTask": "npm",
			"request": "launch",
			"type": "extensionHost"
		},
		{
			"name": "Run Extension",
			"type": "extensionHost",
			"request": "launch",
			"args": [
				"--extensionDevelopmentPath=${workspaceFolder}"
			],
			"outFiles": [
				"${workspaceFolder}/out/**/*.js"
			],
			"preLaunchTask": "${defaultBuildTask}"
		},`
		
What else can I do?

**Comment by r3m0t** [comment]
Check every output channel in the Output window in Extension Development Host for errors.

**Reply by pablojakub** [reply]
When I use 'Run Extension' - default launch created by yo there are no errors.

When I try to run in Launch Extension (created by me - VS Code Extension Development) the error occurs: 
![image](https://github.com/microsoft/vscode-discussions/assets/68683285/dabfb03a-84d4-42ea-9b1b-40a310e9d0d6)

When I click cebug anyway - no errors in any output chanel. When I choose configure task I tried these:
![image](https://github.com/microsoft/vscode-discussions/assets/68683285/29949cdd-d624-4800-8419-a551995c7110)

But prompt still shows

**Reply by r3m0t** [reply]
The preLaunchTask should be "npm: watch".

**Reply by pablojakub** [reply]
Yes and I choose npm:watch. In tasks I have: 
`"tasks": [
		{
			"type": "npm",
			"script": "watch",
			"problemMatcher": "$tsc-watch",
			"isBackground": true,
			"presentation": {
				"reveal": "never"
			},
			"group": {
				"kind": "build",
				"isDefault": true
			}
		},
		]`
		
But still when I press F5 on extension.ts file it doesn't see it and return me back to this selection:
![image](https://github.com/microsoft/vscode-discussions/assets/68683285/155fd422-56bf-444f-847a-89dc54b104ee)

Should I add something more to tasks?

**Reply by pablojakub** [reply, marked as answer]
Ok never min :) 
It turns out that my extension required 1.83.0 vs code version and I had 1.7.x :). I bumped version and change it in package.json and now it's fine: 
![image](https://github.com/microsoft/vscode-discussions/assets/68683285/6ffc5519-1c30-48e8-8b49-77007ae82d16)

Thanks for support :) This ticket can be closed
